### ISSUE TITLE
Support Managing AI APIs within the API Manager via APICTL

### ISSUE THREAD

**Issue by ashera96** [wso2/api-manager#3162]
### Description

- [x] Import AI API support
- [x] Export AI API support
- [x] Import AI-specific (token-based) policy
- [x] Export AI-specific (token-based) policy

### Affected Component

APICTL

### Version

4.4.0

**Comment by ashera96** [comment]
#### [Progress Update]

- API import fails with the following error when running the ctl import API command:

```
➜  ~ apictl import api -f /Users/ashera/.wso2apictl/exported/apis/dev/MistralAIAPI_0.0.2.zip -e prod --verbose
Executed ImportExportCLI (apictl) on Sun, 06 Oct 2024 22:51:16 +0530
[INFO]: Insecure: false
[INFO]: api called
[INFO]: connecting to https://localhost:9444/oauth2/token
[INFO]: Resolving for API path...
[INFO]: API Location: /Users/ashera/.wso2apictl/exported/apis/dev/MistralAIAPI_0.0.2.zip
[INFO]: Creating workspace
[INFO]: Extracting /Users/ashera/.wso2apictl/exported/apis/dev/MistralAIAPI_0.0.2.zip to /var/folders/b5/3ywkk2nn2y54l17zj0863twm0000gn/T/apim1768527064
[INFO]: Substituting environment variables in API files...
[INFO]: Creating the project artifact /var/folders/b5/3ywkk2nn2y54l17zj0863twm0000gn/T/project-artifact1318025712.zip
[INFO]: Creating: MistralAIAPI-0.0.2/
[INFO]: Creating: MistralAIAPI-0.0.2/Definitions/
[INFO]: Creating: MistralAIAPI-0.0.2/Definitions/swagger.yaml
[INFO]: Creating: MistralAIAPI-0.0.2/api.yaml
[INFO]: Creating: MistralAIAPI-0.0.2/api_meta.yaml
[INFO]: Import URL: https://localhost:9444/api/am/publisher/v4/apis/import?preserveProvider=true&rotateRevision=false
Response : {"code":900907,"message":"Error while reading meta information from the definition","description":"Error while reading meta information from the definition","moreInfo":"","error":[]}Error importing API.
Status: 400
Response: {"code":900907,"message":"Error while reading meta information from the definition","description":"Error while reading meta information from the definition","moreInfo":"","error":[]}
[INFO]: Deleting /var/folders/b5/3ywkk2nn2y54l17zj0863twm0000gn/T/project-artifact1318025712.zip
[INFO]: Deleting /var/folders/b5/3ywkk2nn2y54l17zj0863twm0000gn/T/apim1768527064/MistralAIAPI-0.0.2
apictl: Error importing API Reason: 400
Exit status 1
```

Following CTL commands work as is:
- apictl export api (--name <name-of-the-api> --version <version-of-the-api> --provider <provider-of-the-api> --environment <environment-from-which-the-api-should-be-exported>) [flags]
- apictl export policy rate-limiting (--type <type-of-the-throttling-policy> --environment <environment-from-which-the-throttling-policies-should-be-exported>) [flags]
- apictl import policy rate-limiting --file <path-to-api> --environment <environment> [flags]

**Comment by ashera96** [comment]
Upon debugging the API import flow; the point of failure was identified as [1]. The reason why it's failing is because the value that reaches this point is already base64encoded. Hence, we have to skip this flow for API import.

![image](https://github.com/user-attachments/assets/fb27c82b-2774-4e77-bb27-a23376d8c3dd)

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java#L615

**Comment by ashera96** [comment]
#### [Progress Update]

- Policy import is fixed in the latest APIM 4.4.0 pack
- API import still fails and we need to identify whether to fix the export flow or import flow. Created the issue [1] to track the fix for this.

1. https://github.com/wso2/api-manager/issues/3200

**Comment by ashera96** [comment]
#### [Progress Update]

- API export and import is working as expected
- AI policy export and import is working as expected
- Integration tests were added for AI API import/export as well as AI policy import/export via https://github.com/wso2/product-apim-tooling/pull/1220
